### PR TITLE
doc: Fix links to contributing guide 

### LINKED ISSUE #10316: Documentation links to https://docs.cilium.io/en/stable/contributing/contributing/ fail
## Bug report

**General Information**

Documentation links to https://docs.cilium.io/en/stable/contributing/contributing/ fail to find the requested page.

**How to reproduce the issue**

1. Navigate to https://github.com/cilium/cilium
2. Open the link to `contributing` - http://docs.cilium.io/en/stable/contributing/contributing/

### PR DESCRIPTION
**Pull request by CybrPunk**
Fixes the link to the contributing guide in the main README.rst and cilium-actions.yaml
The previous link pointed to a folder structure that had changed recently.
Fixes: #10316

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10322)

### CODE REVIEW

File: .github/cilium-actions.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ move-to-projects-for-labels-xored:
       column: "Backport done to v1.5"
 require-msgs-in-commit:
   - msg: "Signed-off-by"
-    helper: "https://docs.cilium.io/en/stable/contributing/contributing/#developer-s-certificate-of-origin"
+    helper: "http://docs.cilium.io/en/stable/contributing/development/contributing_guide/#developer-s-certificate-of-origin"
```
Comment (pchaigno) [inline .github/cilium-actions.yml]: Any reason you switched to `http`?

**Comment by pchaigno** [comment]
test-docs-please

File: .github/cilium-actions.yml (hunk 1 above)
Comment (CybrPunk) [reply .github/cilium-actions.yml]: hadn't realized I did, I followed a link from the readme which uses http, copied the whole url and pasted it....

Comment (pchaigno) [reply .github/cilium-actions.yml]: Could you please revert this change and sign off your commit?

**Review by joestringer — COMMENTED** [review]
FYI, since apparently those docs don't point it out:

If you `git config user.name "John Doe"` and `git config user.email "john@doe.com"` then when committing the code you do `git commit -s` (or `git commit --amend -s`), the signoff line will be populated by git.

I think it'd be worthwhile to add this to the instructions too.

**Review by jrajahalme — APPROVED** [review]
I'd prefer these four commits to be squashed together to ease backporting (if needed).

**Comment by joestringer** [comment]
test-docs-please

**Pull request merged**